src/west/configuration.py

```diff
@@ -70,19 +70,38 @@ def parse_key(dotted_name: str):
         return section_child
 
     @staticmethod
-    def from_path(path: Path | None) -> '_InternalCF | None':
-        return _InternalCF(path) if path and path.exists() else None
-
-    def __init__(self, path: Path):
-        self.path = path
+    def from_path(path: Path | list[Path] | None) -> '_InternalCF | None':
+        if not path:
+            return None
+
+        # Normalize to a list of existing paths
+        paths = path
+        if isinstance(paths, Path):
+            paths = [Path(p) for p in str(paths).split(os.pathsep)]
+        if not all([p.is_absolute() for p in paths]):
```

Comment:
The `all` can be optimized to use a generator expression instead.

```suggestion
        if not all(p.is_absolute() for p in paths):
```